en/developer/customization/articles/50-liferay-forms/01-form-field-types/00-intro.markdown

```diff
@@ -0,0 +1,112 @@
+---
+header-id: form-field-types
+---
+
+# Form Field Types
+
+[TOC levels=1-4]
+
+The Forms application contains many highly configurable field types
+out-of-the-box. Most use cases are met with one of the existing field types. 
+
+![Figure 1: The Forms application has useful out-of-the-box field types, but you can add your own if you need to.](../../../images/forms-field-types.png)
+
+If you're reading this, however, your use case probably wasn't met with the
+default field types. For example, perhaps you need a color picker field. You
+could create a select field that lists color names. Some users don't, however,
+know that *Gamboge* is the color of spicy mustard (maybe a little darker), and
+anyway, seeing colors is better than reading their names, so you can create
+a field that shows colors. Or, for instance, maybe you need to add a [slider](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range) to measure the level of satisfaction of a product.
+
+These tutorials show you how to
+
+- create a module that adds a *Slider* form field type
+
+- add custom configuration options to your field types
+
+<!-- TODO: upload zip file and update its path below. -->
+
+| **Example project:** The source code for the example *slider* project developed in
+| these tutorials can be downloaded for your convenience. Click
+| [here](https://portal.liferay.dev/documents/113763090/114000653/dynamic-data-mapping-type-slider.zip) to begin downloading the source code zip file.
+
+
+Before getting started, learn the structure of a form field type.
+
+## Anatomy of a Field Type Module
+
+All form field type modules have a similar structure. Here's the directory
+structure of the `dynamic-data-mapping-type-slider` module developed in these
+tutorials:
+
+    .babelrc
+    .npmbundlerrc
+    bnd.bnd
+    build.gradle
+    package-lock.json
+    package.json
+    src
+    └── main
+        ├── java
+        │   └── com
+        │       └── liferay
+        │           └── dynamic
+        │               └── data
+        │                   └── mapping
+        │                       └── form
+        │                           └── field
+        │                               └── type
+        │                                   └── slider
+        │                                       ├── SliderDDMFormFieldTemplateContextContributor.java
+        │                                       ├── SliderDDMFormFieldType.java
+        │                                       └── SliderDDMFormFieldTypeSettings.java
+        └── resources
+            ├── content
+            │   └── Language.properties
+            └── META-INF
+                └── resources
+                    ├── Slider.es.js
+                    ├── Slider.soy
+                    └── SliderRegister.soy
+
```

Comment:
@natocesarrego is this still accurate? In Evan Thibodeau's form field I only see the es.js file in `src/main/resources/META-INF/resources`: See [here](https://github.com/lfrsales/character-limited-text-input-ddm-form-field/tree/master/modules/character-limited-text/src/main/resources/META-INF/resources)